miden-lib/src/transaction/mod.rs

```diff
@@ -124,21 +135,35 @@ impl TransactionKernel {
     ///
     /// The initial stack is defined:
     ///
-    /// > [BLOCK_HASH, acct_id, INITIAL_ACCOUNT_HASH, INPUT_NOTES_COMMITMENT]
+    /// ```text
+    /// [
+    ///     BLOCK_HASH,
+    ///     acct_id,
+    ///     INITIAL_ACCOUNT_HASH,
+    ///     INPUT_NOTES_COMMITMENT,
+    ///     kernel_procs_len,
+    ///     KERNEL_HASH
+    /// ]
```

Comment:
Actually, thinking more about this, we may be able to avoid putting `KERNEL_HASH` into stack inputs at all. Instead, we could add a field to the `BlockHeader` - e.g., something like `kernel_root` which would be a sequential hash of all allowed kernels (for now, we'd have just a single kernel).

Then, during prologue, we'd "unhash" `kernel_root` into memory and the would pick one of the kernels from it for a transaction. This all can be done with advice inputs only, but would require modifying the current `BlockHeader` (how to populate it with the right kernel hash may be a bit tricky - but I think it should be solvable).